assets/js/base/components/cart-checkout/payment-method-icons/utils.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * For an array of icons, normalize into objects and remove duplicates.
+ *
+ * @param {Array} icons Array of icon objects or string based ids.
+ */
+export const normalizeIconConfig = ( icons ) => {
+	const normalizedIcons = {};
+
+	icons.forEach( ( raw ) => {
+		let icon = {};
+
+		if ( typeof raw === 'string' ) {
+			icon = {
+				id: raw,
+				alt: raw,
+				src: null,
+			};
+		} else {
+			icon = {
+				id: raw.id,
+				alt: raw.alt,
+				src: raw.src,
```

Comment:
There's no validation that `id`, `alt`, and `src` exist here right? probably should only set this if the icon is a valid object.